Sources/Networking/Operations/PostReceiptDataOperation.swift

```diff
@@ -39,11 +39,16 @@ final class PostReceiptDataOperation: CacheableNetworkOperation {
         customerInfoCallbackCache: CallbackCache<CustomerInfoCallback>,
         offlineCustomerInfoCreator: OfflineCustomerInfoCreator?
     ) -> CacheableNetworkOperationFactory<PostReceiptDataOperation> {
+        // Can't compute offline CustomerInfo when purchasing non-consumables
+        let offlineCreator: OfflineCustomerInfoCreator? = postData.isNonSubscriptionProduct
```

Comment:
Hmm so this will only work when posting a receipt right? I don't think this will handle this scenario:
- User purchases a consumable while offline (the post fails)
- For some reason, they invalidate customer info or uninstall/reinstall (so no customer info cached)
- They restart the app
- During restart, get customer info request fails with a 5xx, so we try to enter offline entitlements even though it has a consumable.

Admittedly, this seems like an uncommon case (users invalidating customer info in the middle), but ideally we would also handle it.

Reply:
Thanks for pointing this out. The new implementation fixes that 👍🏻 
Also I think it's much cleaner moving the check to use `StoreKit.Transaction.unfinished`.